yarn-project/noir-contracts/src/contracts/test_contract/src/test_contract_interface.nr

```diff
@@ -0,0 +1,145 @@
+/* Autogenerated file, do not edit! */
+  
+use dep::std;
+use dep::aztec::context::Context;
+use dep::aztec::constants_gen::RETURN_VALUES_LENGTH;
+
+struct AStructTestCodeGenStruct {
+  amount: Field,
+  secretHash: Field,
+}
+
+struct ADeepStructTestCodeGenStruct {
+  aField: Field,
+  aBool: bool,
+  aNote: ANoteADeepStructTestCodeGenStruct,
+  manyNotes: [ManyNotesADeepStructTestCodeGenStruct;3],
+}
+
+struct ANoteADeepStructTestCodeGenStruct {
+  amount: Field,
+  secretHash: Field,
+}
+
+struct ManyNotesADeepStructTestCodeGenStruct {
+  amount: Field,
+  secretHash: Field,
+}
+
+struct TestContractInterface {
+  address: Field,
+}
+
+impl TestContractInterface {
+  fn at(address: Field) -> Self {
+      Self {
+          address,
+      }
+  }
+  
+  fn createL2ToL1MessagePublic(
+    self,
+    context: &mut Context,
+    amount: Field,
+    secretHash: Field
+  ) {
+    let mut serialised_args = [0; 2];
+    serialised_args[0] = amount;
+    serialised_args[1] = secretHash;
+
+    context.call_public_function(self.address, 0x1c031d17, serialised_args)
+  }
+  
+
+  fn createNullifierPublic(
+    self,
+    context: &mut Context,
+    amount: Field,
+    secretHash: Field
+  ) {
+    let mut serialised_args = [0; 2];
+    serialised_args[0] = amount;
+    serialised_args[1] = secretHash;
+
+    context.call_public_function(self.address, 0x0217ef40, serialised_args)
+  }
+  
+
+  fn getPortalContractAddress(
+    self,
+    context: &mut Context,
+    aztec_address: Field
+  ) -> [Field; RETURN_VALUES_LENGTH] {
+    let mut serialised_args = [0; 1];
+    serialised_args[0] = aztec_address;
+
+    context.call_private_function(self.address, 0xe5df1726, serialised_args)
+  }
+  
+
+  fn getPublicKey(
+    self,
+    context: &mut Context,
+    address: Field
+  ) -> [Field; RETURN_VALUES_LENGTH] {
+    let mut serialised_args = [0; 1];
+    serialised_args[0] = address;
+
+    context.call_private_function(self.address, 0x553aaad4, serialised_args)
+  }
+  
+
+  fn getThisAddress(
+    self,
+    context: &mut Context
+  ) -> [Field; RETURN_VALUES_LENGTH] {
+    let mut serialised_args = [0; 0];
+
+    context.call_private_function(self.address, 0xd3953822, serialised_args)
+  }
+  
+
+  fn getThisPortalAddress(
+    self,
+    context: &mut Context
+  ) -> [Field; RETURN_VALUES_LENGTH] {
+    let mut serialised_args = [0; 0];
+
+    context.call_private_function(self.address, 0x82cc9431, serialised_args)
```

Comment:
There's a `call_private_function_no_args` method in `Context`. I wonder if that needs to be called if there are no args being passed. The difference appears to be:
- if you call `call_private_function` and pass a `0`, it'll hash that `0` into an args hash.
- if you call `call_private_function_no_args`, it'll pass `0` as the args hash.

@sirasistant what is the best approach? Is it important to call `call_private_function_no_args` for functions which don't have any args (except for the `PrivateContextInputs`)?

There are a few functions in `test_contract/src/main.nr` which don't have any args (except sometimes the `PrivateContextInputs`).

Reply:
> if you call call_private_function and pass a 0, it'll hash that 0 into an args hash.

But we're not passing a 0 here, we're passing an empty array, which eventually calls arguments::pack, which returns zero from an empty set (see [here](https://github.com/AztecProtocol/aztec-packages/blob/fc80ef38981b6ecf4c954726e93f03c2b3a1bfc2/yarn-project/acir-simulator/src/packed_args_cache.ts#L49-L51)). Still, I'll add a test to make sure it works fine.

Reply:
We shouldn't be using the result of `arguments::pack` in the circuit. It's the result of `hash_args` we should be looking at (which seems like it would return the hash of an array of zeroes?). There's a discussion in a separate issue to change this and to remove the assertion `assert(args_hash == arguments::pack_arguments(args));`. (These assertions should be done in typescript to error early; they're not assertions which provide meaningful circuit constraints.

Reply:
Looks like you've updated the hash_args logic to handle this edge case - thank you :)
https://github.com/AztecProtocol/aztec-packages/pull/1487/commits/71556c68bdd434a1022d725bd82863bd1008bf99